modules/notify/main/index.coffee

```diff
@@ -1,33 +1,32 @@
 # utility method for setting up notifications
 setupNotification = (notification, selection) ->
-  if notification.options.icon? and notification.options.icon.length > 0
-    selection.append('div')
-      .class('hx-notification-icon-container')
-      .append('i')
-        .class('hx-notification-icon ' + notification.options.icon)
+  icon = if notification.options.icon? and notification.options.icon.length > 0
+    hx.detached('div').class('hx-notification-icon-container')
+      .add(hx.detached('i').class('hx-notification-icon ' + notification.options.icon))
 
-  selection
-    .append('div')
-      .class('hx-notification-text')
-      .text(notification.message)
+  content = hx.detached('div').class('hx-notification-content')
+  notification.options.renderer(content.node(), notification.message)
 
   if notification.options.pinnable
-    notification.domPin = selection
-      .append('div')
-        .class('hx-notification-icon-container hx-notification-pin')
-        .on 'click', 'hx.notify', -> togglePin(notification)
-
-    notification.domPin.append('i')
-      .attr('class', 'hx-icon hx-icon-thumb-tack')
+    pin = hx.detached('div')
+      .class('hx-notification-icon-container hx-notification-pin')
+      .on('click', 'hx.notify', -> togglePin(notification))
+      .add(hx.detached('i').attr('class', 'hx-icon hx-icon-thumb-tack'))
 
+    notification.domPin = pin
     updatePinnedStatus(notification)
 
+  close = hx.detached('div')
+    .class('hx-notification-icon-container hx-notification-close')
+    .on('click', 'hx.notify', -> notification.close())
+    .add(hx.detached('i').class('hx-icon hx-icon-close'))
+
+
   selection
-    .append('div')
-      .class('hx-notification-icon-container hx-notification-close')
-      .on 'click', 'hx.notify', -> notification.close()
-      .append('i')
-        .class('hx-icon hx-icon-close')
+    .add(icon)
+    .add(content)
+    .add(pin)
+    .add(close)
```

Comment:
I updated how the notification is built because it was a little confusing.
All the elements are now built detached and added to the notification at the end.